cpp/ql/src/semmle/code/cpp/ir/implementation/raw/internal/TranslatedElement.qll

```diff
@@ -49,6 +49,13 @@ private Element getRealParent(Expr expr) {
   )
 }
 
+/**
+ * Holds if `expr` is a constant of a type that can be replaced directly with
+ * its value in the IR. This does not include address constants as we have no
+ * means to express those as QL values.
+ */
+predicate isIRConstant(Expr expr) { exists(expr.getValue()) }
```

Comment:
This looks like a reasonable change for the IR. Can this predicate be `private`?

Reply:
It can't be private: it's also used in `TranslatedExpr.qll`.

Reply:
Makes sense. All of the IR construction modules are imported privately by their consumers anyway.